### PR TITLE
Add Vekna ritual for nighttime PR check

### PR DESCRIPTION
**Pull request by fancysnake**
Adds `pr_check`, a [vekna](https://github.com/fancysnake/vekna) ritual that
takes every open pull request of yours in turn, oldest-modified first, and
leaves each one ready to look at in the morning:

    vekna cast pr_check [--bound N]

Per pull request: merge the base branch in, make `mise run devcheck` green,
close the `mise run diff-cover` gap, post a thermo-nuclear quality review if
none is there, then triage the open review comments. A branch ends the night
either labelled `pr::qa` with a `qa.md` of manual scenarios, or carrying a
`triage.md` saying what still has to be done. Anything this run put in front of
you to read also earns `pr::cr`.

**Nothing is ever pushed.** The report says which branches are waiting for
that, and pushing stays yours.

## How it holds together

It runs unattended, so it asks nothing — at 3am a permission prompt is a hang.
That is a deliberate break with vekna's usual bargain, where spending another
agent attempt is a `decide`. Per-step, per-branch budgets take that decision
instead: a step may be retried `--bound` times, going green clears its count,
and moving to the next pull request clears them all. A step that runs out gives
up on that pull request only — the worktree is released into a named stash the
report points at, the branch is reported blocked, and the next one starts.

Agents stay non-deterministic inside a step and deterministic between them.
Every step re-reads git or `gh` rather than believing what the agent said it
did, and every prompt forbids the shortcuts by name: no disabling a lint rule,
no `noqa`, no `type: ignore`, no deleting the failing test, no lowering a
threshold.

Two things end the whole run rather than one branch: a worktree that is not
clean, and an agent that dies mid-flight. Both print the report first, then
fail the cast.

## Notes for review

**`edges/` is the right home.** Nothing in `ludamus` imports `edges`, and
import-linter's `inside-edges` contract passes — it is where independent
scripts live. That does mean a dev-only dependency is imported from inside the
package, so the exemption is declared module-scoped rather than rule-wide:

```toml
DEP004 = ["debug_toolbar", "vekna"]
```

**`broad-exception-caught` is now disabled repo-wide, on purpose.** Ruff's
`blind-except` already checks it, per-site and with a working suppression
directive. Keeping both mechanisms meant the inline pylint comments in
`gates/web/django/urls.py` and `links/db/django/repositories/storage.py` were
suppressing a rule that no longer fires, so this PR deletes them. The site
idiom was tried first and does not survive here: black reflows the trailing
comment onto the wrong line once the handler binds `as error`.

**No tests, deliberately.** `[tool.coverage.run] omit` covers
`src/ludamus/edges/*`, so these lines sit outside the coverage gate by design.
The pure helpers — the report rendering, the budget quartet — move to a
submodule that coverage does measure once vekna supports module rituals
(`0.4.0`), and get tested there.

**tingle is +1**, a second `blind-except` suppression. `_ask` is two functions
now so that neither half's return type carries a payload nobody reads, and both
halves still have to survive a killed CLI.

No UI, so no screenshots.

## Summary by CodeRabbit

* **New Features**
  * Added an automated pull-request maintenance workflow that processes pending work sequentially.
  * Handles branch synchronization, merge conflicts, gate repairs, coverage checks, review comments, QA preparation, labeling, and final status reporting.
  * Added safeguards for dirty worktrees, failed steps, retry limits, and independently blocked pull requests.
* **Documentation**
  * Documented a known issue affecting task execution from dependency worktrees and the required trust configuration.
* **Chores**
  * Updated development checks, linting, and project tooling configuration, including stricter coverage validation.

### CODE REVIEW

**Comment by fancysnake** [comment]
## Thermo-nuclear code quality review

846 new lines, one file, no tests. The prose is unusually good — most of the
comments explain a decision rather than restate the line under them, and the
state machine is genuinely easy to follow. The problems are almost all about
*where this lives* and *what it promises versus what it does*, not about the
step graph itself.

---

### 1. `src/ludamus/edges/` is the wrong home, and CI already says so

`mise run dep:audit:py` **fails on this branch**:

```
src/ludamus/edges/rituals.py:36:1: DEP004 'vekna' imported but declared as a dev dependency
… ×4
Found 4 dependency issues.
```

`.github/workflows/audit.yml` triggers on `pyproject.toml` and `poetry.lock`,
both of which this PR touches, and the job is not `continue-on-error`. So this
is a red required check, not a style opinion.

The underlying problem is real, not a deptry quirk: a dev-only dependency is
imported from a module inside the shipped `ludamus` package. Anything that
walks the package in a production install — and any future `import
ludamus.edges.rituals` — hits `ModuleNotFoundError: vekna`.

Two more things fall out of the same placement:

- `[tool.coverage.run] omit` lists `src/ludamus/edges/*`. 846 lines of new
  logic land **outside the coverage gate entirely**. That is why `diff-cover`
  has nothing to say about this PR.
- It needed a `vulture` `ignore_names` entry for `pr_check`, which only exists
  because the symbol is inside the scanned package.

The move: `rituals/pr_check.py` at the repo root, or under `scripts/`. It is a
one-line change in `.vekna.toml` (`files = [...]`), and it deletes the deptry
failure, the coverage exemption, the vulture ignore, and the "does a ritual
belong in `edges`?" question all at once. That is the code-judo move on this
PR — one path change removing four separate accommodations.

### 2. No tests, in the ritual whose job is closing coverage gaps

`CLAUDE.md`'s definition of done asks for happy path plus one edge case. The
coverage `omit` above means nothing enforces it here, which makes it easier to
skip, not more acceptable to.

There is a lot of genuinely pure, genuinely testable surface that needs no
vekna runtime at all: `_report` / `_summary` / `_line` / `_names`, `_counted`,
the budget quartet (`_spent` / `_exhausted` / `_charged` / `_cleared`),
`_release` / `_commit` / `_label`, `_said`. One test module over those covers
most of the logic a reader would actually want pinned — particularly the
budget lifecycle, which is the thing this ritual's correctness rests on and the
thing a future edit is most likely to get subtly wrong.

### 3. Half of `_ask`'s return type is dead

Every one of the six untyped call sites does exactly this and nothing else:

```python
said = await _ask(...)
if isinstance(said, Fallen):
    return goto(report, _abandoned(work, said.reason))
```

`said`'s `str` payload is never read anywhere in the file. So the non-`output`
overload's real contract is `Fallen | None`, and the `str` in the union is
carrying no information — it just makes every call site's type wider than the
thing it is used for.

Split it into two named functions — one returning `Fallen | None`, one
returning `OutputT | Fallen` — and **both `@overload` stubs disappear** along
with the `reply.text` unwrap. That is ~20 lines and one type-level concept gone
for no behaviour change.

### 4. `Fallen` conflates two failures, and the docstring only admits one

The module docstring says the run ends on "an agent that dies mid-flight (a
spent token budget, a killed CLI)". But `_ask` catches bare `Exception`, and
vekna raises `CodingOutputError` when the agent's JSON does not validate
against the requested schema, plus `CodingSessionError` for a malformed
thread declaration. Both become `Fallen`, and `Fallen` ends the **whole night**.

So one badly-formed `TriageNotes` from the third PR kills the run for the
fourth, fifth and sixth — reported as "the agent stopped mid-flight", which is
not what happened. A schema miss is per-branch and recoverable; it should route
to `set_aside` like every other per-branch giving-up.

Catching `CodingOutputError` / `CodingSessionError` explicitly and returning a
distinct value also narrows what the blind `except Exception` has to justify —
which is directly relevant to §5.

### 5. The global pylint disable, and the two suppressions it orphans

```diff
-        missing-docstring
+        missing-docstring,
+        # Checked by ruff
+        broad-exception-caught
```

`CLAUDE.md`: *"Don't ignore lint rules globally."* This turns off a real check
for the entire codebase to accommodate one new call site.

The repo already has the canonical per-site idiom for exactly this, in
`src/ludamus/gates/web/django/urls.py`:

```python
Exception  # ruff:ignore[blind-except]  # pylint: disable=broad-exception-caught
```

Worse, the global disable leaves two now-dead inline suppressions behind —
that one, and `src/ludamus/links/db/django/repositories/storage.py`. Whichever
way this goes, the PR should not leave both mechanisms live:

- reuse the established site idiom and drop the `.pylintrc` change, or
- keep the global change on the "ruff already checks it" argument, **delete the
  two orphaned inline comments**, and say so in the PR body.

(I verified the ruff side works: removing `# ruff: ignore [blind-except]` makes
`mise run lint:ruff` report `blind-except` at that line. The directive is real,
including with the space before the bracket.)

`mise run lint:tingle` is red at **+3** — `ruff-comment`, `pylint-disables`,
and a `type-object` hit that is a false positive on the literal word "object"
in the comment above `class Work`. Tingle is advisory in CI, so this is not a
blocker, but one of the three is free: reword that comment.

### 6. Two string-grep boundaries where the file already knows better

```python
if _THERMO_TITLE in seen.stdout:
```

`seen.stdout` is the **raw JSON** of every comment on the PR. Any comment that
quotes the heading — a human replying to the review, a bot echoing it, a
follow-up that pastes the title — makes this step believe a review already
exists and silently skip posting one. That is a false negative in the step
whose entire job is "was the review posted?", and it is decided by substring
luck.

The file already demonstrates the right move eleven lines up: `_PULLS =
TypeAdapter(list[PullRequest])`. Parse the comments the same way and match the
first line of each body against `_THERMO_TITLE`, which is exactly what the
prompt in `_thermo` asks the agent to produce.

Same class, weaker case: `if "Missing" in output` in `cover`. Harder to avoid
against diff-cover's text output, so I would leave it — but the comment
defending it should say it is a deliberate compromise rather than reading as
the intended design.

### 7. `_ahead()` is asked on the wrong HEAD in one reachable path

`sync_branch` routes to `set_aside` when `git checkout <branch> && git merge
--ff-only origin/<branch>` fails. If the failure was the **checkout**, HEAD is
still on `base`. `set_aside` then calls:

```python
unpushed=await _ahead(work.pr.branch)   # git rev-list --count origin/<pr-branch>..HEAD
```

with HEAD on `base`, so this counts base commits absent from the PR branch and
reports a confident, wrong "N unpushed" for that PR — in a report whose whole
purpose is telling you what still needs pushing. The `unpushed: int | None`
boundary is otherwise handled really well (the "we could not tell" vs "nothing
to do" distinction in `_report` is exactly right); this path should return
`None` rather than a number, e.g. by checking `git rev-parse --abbrev-ref HEAD`
matches the branch first.

### 8. The report does not point at the stash it promises

`_release`'s comment:

> A conflicted merge goes back where it was; the rest goes into a named stash
> **the report points at**.

`_line` prints number, branch, outcome, unpushed and note. The stash label
(`pr_check left <branch> unfinished`) appears nowhere in the report. At 3am the
whole value of naming the stash is that the morning report tells you it exists.
Either put it in the `Checked.note` on the release path, or drop the claim.

### 9. The four repair loops restate the same policy four times

`resolve_conflicts`, `gate_check`, `cover` and `quality_review` are the same
shape: measure, green → `_cleared` + advance, exhausted → `set_aside` + note,
otherwise ask → `Fallen` → abandon, else `_charged` + re-enter self.

I checked the constraint you documented and it holds — vekna's
`_mills/graph.py` reads `goto` targets off `ast.Name` args, so a target named
inside a helper really is an edge `rituals show` loses. The four `goto`s have
to stay written out.

But the *decision* does not. A helper returning a small tagged result, with the
four `goto`s as a flat match at the call site, keeps every edge visible while
stating the budget policy once. Medium conviction — if it does not come out
shorter, drop it. The unconditional win is smaller: the `isinstance(said,
Fallen) → goto(report, _abandoned(...))` pair is verbatim six times and is pure
policy, not per-step logic.

### 10. Smaller

- `_modified` is a one-line identity wrapper used exactly once, as a sort key.
  `key=attrgetter("updated_at")` and delete it. (`_q` earns its keep at ten
  call sites; this one does not at one.)
- `_run` / `_work`: `None` means "keep the old value", so `stopped=""` and
  `note=""` are unrepresentable as updates. Nothing needs to clear a field
  today, so this is fine — but it is the kind of thing that bites silently
  later. Worth one line of comment on the sentinel, since the rest of the file
  is so careful about exactly this class of trap.
- Permission asymmetry: `_READING` is the only call that declares an allowlist,
  and it is the only **read-only** one. The six that rewrite the working tree
  pass `opts=None`, and vekna resolves that to `bypassPermissions` (`"default"
  if gate is not None else "bypassPermissions"`). That is probably what you
  want for an unattended run, but the comment on `_READING` reads as though it
  is the tightly-scoped exception among constrained calls, when it is the only
  constrained call at all. Say it plainly.
- `_DEVCHECK` / `_COVERAGE` do not include `lint:tingle`, so the ritual's
  definition of green is narrower than the repo's `fullcheck`. This PR is the
  first branch that is green by the ritual's standard and red by the repo's.

### File size

846 lines, so no threshold crossed — but the seam is already visible (prompts
and report rendering versus the state machine), and the second ritual in this
file pushes it over 1k. Not a blocker now; worth deciding before then.

---

### Papercut

Running any `mise run <task>` with cwd inside `.venv/src/vekna` fails with
`Config files in .../vekna/mise.toml are not trusted` instead of running the
repo's task. The git-sourced dependency this PR adds ships its own `mise.toml`
into the venv, and mise's config discovery walks up into it. Worth a
`mise trust` note in the sandbox docs, or logging via `mise run papercut`.

---

### Verdict

Request changes. §1 is a red required check and drives most of the rest — fix
the placement and the coverage exemption, tests, and the deptry failure all
resolve together. §3 and §4 are cheap deletions that make the contract match
what the module already says about itself. §7 is a wrong number in the one
artifact a human reads in the morning.

**Comment by fancysnake** [comment]
Follow-up on the review above, after discussion with the author — two items
resolve differently than written.

**§1, the file's location: withdrawn.** `edges` is where independent scripts
live, and the import direction backs that: nothing in `ludamus` imports
`edges`, and `import-linter`'s `inside-edges` contract passes. The file stays
put.

What does *not* resolve on its own is the CI failure it caused. `mise run
dep:audit:py` is red on this branch and is a blocking check here (`audit.yml`
fires on `pyproject.toml` and `poetry.lock`, both touched, no
`continue-on-error`):

```
src/ludamus/edges/rituals.py:36:1: DEP004 'vekna' imported but declared as a dev dependency
… ×4
```

Keeping the file in `edges` means declaring the exemption rather than moving
out of its way. The repo already has the precedent, so this is a one-line
change and the check goes green:

```toml
[tool.deptry.per_rule_ignores]
DEP004 = ["debug_toolbar", "vekna"]
```

Module-scoped, not rule-wide — it says "this one dev dependency is imported
from a script that never runs in production", which is exactly the claim.

**§2, tests: deferred, deliberately.** `[tool.coverage.run] omit` covers
`src/ludamus/edges/*`, so these lines sit outside the coverage gate by design
rather than by accident. The plan is to revisit once vekna supports module
rituals, at which point the pure helpers — `_report` / `_summary` / `_line`,
`_counted`, the budget quartet — can move to a submodule that coverage does
measure. Noting it here so the gap is on the record as a decision with a
follow-up, not an oversight.

Everything else in the review stands as written: §3 (dead half of `_ask`'s
return type), §4 (`Fallen` conflating a dead CLI with a schema miss), §5 (the
global pylint disable and its two orphaned suppressions), §6 (`_THERMO_TITLE in
seen.stdout` grepping raw JSON), §7 (`_ahead()` measured on the wrong HEAD),
§8 (the stash the report promises but never names).

**Comment by fancysnake** [comment]
## Review resolved

Every item below, with what happened to it. Four commits: `ffce9aa4`,
`e40d8028`, `ea6ee4b6`, `6a60ce72`.

### Fixed

**§1 — the CI failure.** Location withdrawn as agreed; the deptry red is
declared rather than moved out of the way, module-scoped:
`DEP004 = ["debug_toolbar", "vekna"]`. `mise run dep:audit:py` is green.

**§3 — the dead half of `_ask`'s return type.** Split into `_ask`
(`Fallen | None`) and `_ask_for` (`OutputT | Fallen | Misread`). Both
`@overload` stubs and the `reply.text` unwrap are gone, and the six call sites
are now `if fallen := await _ask(...)`.

**§4 — `Fallen` conflating two failures.** `_ask_for` catches
`CodingOutputError` ahead of the blind clause and answers `Misread`, which
`read_comments` routes to `set_aside`. A schema miss now costs one pull request
instead of every one behind it.

The `CodingSessionError` half is declined: it is raised only for a `session`
that is not a `Session` member or a `key` that is empty after stripping, and
`_ask`/`_ask_for` can produce neither. Unreachable from this file.

**§6 — `_THERMO_TITLE in seen.stdout`.** Parsed with a `Comments` model and
matched against the first line of each body, which is what `_thermo` asks the
agent to write. A comment that merely quotes the heading no longer counts as
the review. (The `"Missing" in output` half stays — diff-cover has no
structured output to parse, as you said.)

**§7 — `_ahead()` on the wrong HEAD.** Folded into the existing shell call:

```sh
test <branch> = "$(git rev-parse --abbrev-ref HEAD)" && git rev-list --count ...
```

A mismatch exits non-zero and falls through to the `None` the function already
returned for "we could not tell", so no new branch in Python.

**§8 — the stash the report never named.** `_stash(branch)` is shared by
`_release` and `set_aside`, and the note carries it. Only on the path that
actually stashed: `git stash push` against a clean tree exits 0 having saved
nothing, so `_release` tests the tree first and echoes its own marker rather
than the report claiming a stash that is not there.

**§10 — the `_READING` comment.** Rewritten to say it plainly: it is the only
constrained call in the file, the other six pass no opts and so run at vekna's
`bypassPermissions` default, and that is the decision an unattended run makes.

**tingle.** Down from +3 to +1. The `type-object` hit was the literal word in a
comment and is reworded. The remaining +1 is a second `blind-except`
suppression — the price of the §3 split, since both halves still have to
survive a killed CLI.

### Resolved differently

**§5 — the global pylint disable.** Kept, and the two orphaned inline
suppressions deleted instead — the second of the two options you offered.

The site idiom does not survive this file. `black` reflows

```python
except (
    Exception  # ruff:ignore[blind-except]  # pylint: disable=broad-exception-caught
) as error:
```

onto the wrong line once `as error` is present, which is why the disable went
global in the first place. So `urls.py` keeps its ruff directive and loses the
parentheses that only existed to fit the deleted half, and `storage.py` needs no
ruff directive at all — `blind-except` exempts a handler that logs with
`exc_info`. One mechanism is live, not two.

### Declined

**§9 — the four repair loops.** You verified the constraint that keeps the four
`goto`s written out, and rated the rest medium conviction. What is left is a
tagged result plus a match at each call site for about the same line count.
The unconditional half came free with §3: the `isinstance(said, Fallen)` pair
is now a single walrus.

**§10 — `_modified`.** False positive. `attrgetter("updated_at")` is
`attrgetter[Any]` and a lambda's parameter is untyped, and this repo's mypy
rejects both:

```
error: Expression type contains "Any" (has type "attrgetter[Any]")  [misc]
error: Expression type contains "Any" (has type "Callable[[Any], Any]")  [misc]
```

The annotated one-liner is the only one of the three shapes that carries a
type. Tried it, reverted it, and left a comment saying why so it does not get
raised again.

**§10 — the `_run`/`_work` sentinel.** Already commented: the block above `_run`
says `None` means "whatever the last one had".

### Deferred

**§2 — tests.** As agreed: `[tool.coverage.run] omit` covers
`src/ludamus/edges/*`, revisit when vekna supports module rituals and the pure
helpers can move to a submodule coverage measures.

**§10 — `_DEVCHECK`/`_COVERAGE` omitting `lint:tingle`.** Real: `fullcheck` is
`devcheck` + `diff-cover` + `lint:tingle`, so the ritual's green is narrower
than the repo's. Left alone deliberately — tingle is advisory, and teaching an
unattended agent to chase a debt counter is a worse failure mode than the gap.

**File size.** 846 lines, no threshold crossed. The prompts / report-rendering
seam is noted for whenever the second ritual arrives.

### Verification

`dep:audit:py`, `lint:black`, `lint:ruff`, `lint:mypy`, `lint:pylint` green;
`devcheck` green at 3603 passed, 9 skipped; `vekna rituals show pr_check` draws
every edge including the new `read_comments → set_aside`.

File: src/ludamus/edges/rituals.py [hunk 1]
```diff
@@ -0,0 +1,916 @@
...
+# Nothing reads what the agent said back: these calls are judged by what they
+# left in the worktree, which the step that follows reads out of git. So the
+# only answer worth returning is whether the agent was still standing.
+async def _ask(
+    prompt: str, *, opts: CodingOpts | None = None, key: str | None = None
+) -> Fallen | None:
+    session = Session.CONTINUE if key is not None else Session.NEW
+    try:
+        await coding(prompt, opts=opts, session=session, key=key)
+    except Exception as error:  # ruff: ignore [blind-except]
```
[the remark this answers is not in the document]
Comment (fancysnake) [reply src/ludamus/edges/rituals.py]: Verified locally: `ruff check src/ludamus/edges/rituals.py` passes clean. `[tool.ruff] preview = true` is set in `pyproject.toml`, which is exactly what makes the named-rule suppression form real — so `RUF102` is not raised and `BLE001` *is* suppressed. Stripping the two directives from a copy of the file makes ruff report `blind-except` at both lines, which confirms they are doing work rather than being ignored. No change needed; resolving.

File: src/ludamus/gates/web/django/urls.py [hunk 2]
```diff
@@ -41,9 +41,7 @@ def healthz(
         with connection.cursor() as cursor:
             cursor.execute("SELECT 1")
         _healthz_cache.update(time=now, ok=True)
-    except (
-        Exception  # ruff:ignore[blind-except]  # pylint: disable=broad-exception-caught
-    ):
+    except Exception:  # ruff:ignore[blind-except]
```
[the remark this answers is not in the document]
Comment (fancysnake) [reply src/ludamus/gates/web/django/urls.py:42]: Same verification as the sibling thread: `ruff check src/ludamus/gates/web/django/urls.py` is clean, because `[tool.ruff] preview = true` makes `# ruff:ignore[blind-except]` a real suppression. `RUF102` is not raised.

On narrowing the catch: the broad clause predates this branch. All this PR does here is drop the now-redundant `# pylint: disable=broad-exception-caught` half, since `.pylintrc` delegates `broad-exception-caught` to ruff — settled with the human reviewer in §5 of the thermo-nuclear review. The breadth is deliberate: a liveness probe should answer 503 for anything the DB round-trip can throw, including driver/socket errors that are not `django.db.Error`. Narrowing it would turn some outages into a 200. Resolving.

File: src/ludamus/edges/rituals.py [hunk 3]
```diff
@@ -0,0 +1,945 @@
...
+_TRIAGE_READ = f"""\
+Read the open, unresolved review comments on this pull request and triage them
+against the code as it stands on this branch right now.
+
+Read the review threads with `gh` — `gh pr view --json reviews,comments` for
+what sits on the pull request itself, and
+`gh api repos/{{owner}}/{{repo}}/pulls/<number>/comments` for the inline
+threads,
+which the first command does not return. Then read the code each comment points
+at.
```
[the remark this answers is not in the document]
Comment (fancysnake) [reply src/ludamus/edges/rituals.py]: Fixed in `e8f9345c`.

Confirmed the premise first — REST review comments carry no resolution state, so the prompt was asking the agent to filter on something it could not see. `_TRIAGE_READ` now hands it a GraphQL call for `reviewThreads` with `isResolved`, `path`, `line` and the comment bodies, and says plainly that a node with `isResolved: true` is settled and gets skipped whole. The query in the prompt is the one I ran against this pull request to check it, verbatim.

`_READING` allows `Bash` without a per-command list, so `gh api graphql` needed no permission change.

File: src/ludamus/edges/rituals.py [hunk 4]
```diff
@@ -0,0 +1,916 @@
...
+        reason = f"could not add the {_QA_LABEL} label: {_said(labelled)}"
+        return goto(set_aside, _work(work, note=reason))
+    if fallen := await _ask(_QA):
+        return goto(report, _abandoned(work, fallen.reason))
+    landed = await shell(_commit("docs: manual test scenarios for this branch"))
+    if landed.exit_code:
+        return goto(
+            set_aside, _work(work, note=f"could not commit qa.md: {_said(landed)}")
+        )
+    return goto(finish_pr, Closed(work=work, outcome="qa"))
```
[the remark this answers is not in the document]
Comment (fancysnake) [reply src/ludamus/edges/rituals.py]: Fixed in `e8f9345c`, with one addition — the reorder alone was not enough.

`mark_qa` is now ask → check → commit → label, so a dead agent no longer leaves `pr::qa` on a branch with no `qa.md`.

The extra check is `test -f qa.md` before the commit. `_commit` is `git add -A && (git diff --cached --quiet || git commit …)`, so it exits 0 with nothing staged — deliberate, because the repair loops above reach it with nothing to say. That means a green commit does not prove the agent wrote the file, and reordering alone would still have let the label make a promise nobody checked. A missing file now routes to `set_aside` with "the agent wrote no qa.md" in the note.

The suggested diff was against `_label(number=…, label=…)`; the helper is variadic (`_label(*labels, number=…)`), so the ordering is what got taken, not the patch.
